FT Edit all accommodations

Cheza-Dzabala opened the pull request:
#### What does this PR do?
- Allows a supplier to edit their accommodations

#### Description of Task to be completed?
- A supplier should be able to edit the following:

- The supplier is able to edit a name for his facility.
-The supplier is able to edit a description of his facility.
- The supplier is able to edit the location (i.e address) for his facility.
- The supplier is able to edit an image/images of his/her facility.
- The supplier is able to edit the space available in his facility.
- The supplier is able to edit the cost of his accommodation facility. 
- The supplier is able to edit on services that his/her accommodation facility offers.
- The supplier is able to edit amenities that his facility offers

#### How should this be manually tested?

1. Clone this repository
2. run `npm install`
3. run `npm run dev`
4. Log in with a supplier account
5. Access the accommodations section of the app
6. Locate an accommodation owned by the supplier and hit edit
7. Update details of the accommodation and you will be redirected to the accommodation page

#### Any background context you want to provide?
N /A
#### What are the relevant pivotal tracker stories?
 #170308209
#### Screenshots (if appropriate)
<img width="1651" alt="Screen Shot 2019-12-19 at 11 45 45" src="https://user-images.githubusercontent.com/13886438/71165790-25838a00-225a-11ea-9390-970fcd75bd21.png">
<img width="1665" alt="Screen Shot 2019-12-19 at 11 46 02" src="https://user-images.githubusercontent.com/13886438/71165791-261c2080-225a-11ea-9994-d58cad5cdb3f.png">
<img width="331" alt="Screen Shot 2019-12-19 at 11 46 26" src="https://user-images.githubusercontent.com/13886438/71165793-261c2080-225a-11ea-93de-c707c48db07c.png">
<img width="1146" alt="Screen Shot 2019-12-19 at 11 46 50" src="https://user-images.githubusercontent.com/13886438/71165796-26b4b700-225a-11ea-8c46-c7b3b27fc181.png">

#### Questions:
N / A

Cheza-Dzabala committed:
FT Edit all accommodations

- A supplier can edit all their accommodations

[finishes #170308209]

nignanthomas reviewed (changes requested):
@Cheza-Dzabala 
Great job bro

Kindly add an `Edit` button also in `Single Accommodation` for the Supplier.
Also, on mobile view (iPad especially) the textareas are too squeezed as you can see here.
Hope you do not mind working on that.
<img width="822" alt="Screen Shot 2019-12-19 at 15 54 17" src="https://user-images.githubusercontent.com/42959540/71179399-c6813d80-2278-11ea-8553-eafef8e499b9.png">

sabin18 reviewed (changes requested):
@Cheza-Dzabala good job! However, it is better to display images horizontally. Work on that.

nakiwuge commented on src/styles/components/editor.scss:
```diff
@@ -0,0 +1,6 @@
+.editor {
+	border: 1px solid #c2c2c2;
```
Move this color to variables

nakiwuge commented on src/styles/components/editor.scss:
```diff
@@ -0,0 +1,6 @@
+	border: 1px solid #c2c2c2;
+	padding: 10px;
+	height: 50px;
+	background-color: white;
```
Move this color to variables

nakiwuge commented on src/tests/accommodationTests/editAccommodationTest.spec.js:
```diff
@@ -0,0 +1,113 @@
+  it('Should show loading page', async () => {
+    const component = setUp(initialState, {})
+    expect(findByTestAttribute(component, 'loading-page').length).toBe(1);
+
```
Remove this line

nakiwuge commented on src/tests/accommodationTests/editAccommodationTest.spec.js:
```diff
@@ -0,0 +1,113 @@
+    })
+    component.setState({ accommodation: accommodationMocks, isLoading: false })
+    component.instance().handleChange({ target: { value: 'this hotel', name: 'name' } });
+    component.instance().updateAccommodation({ target: { ame: 'name', }, preventDefault: jest.fn() });
```
Add an expect

Cheza-Dzabala replied:
Fixed this, not sure why the `outdated` tag isn't pasted on it.

nakiwuge commented on src/views/accommodations/editAccommodation.js:
```diff
@@ -0,0 +1,213 @@
+
+    for (const key in data) {
+      if (data[key]) {
+        console.log(key);
```
Remove console

nakiwuge reviewed (changes requested):
@Cheza-Dzabala  good job on the implementation. Kindly work on the feedback below
- Move the image update to the bottom of the form it does not look good on the top.
- Add a `cancel` button. This button should redirect the user back to the accommodation they were editing.
- Make the `Edit` button on `viewing single accommodation` a link and move it to the extreme right preferably in line with the breadcrumbs
- Raise your coverage to 90+ and check out the other comments I have left on this pr

sabin18 reviewed (approved):
Good job @Cheza-Dzabala .

Cheza-Dzabala commented:
> @Cheza-Dzabala good job on the implementation. Kindly work on the feedback below
> 
> * Move the image update to the bottom of the form it does not look good on the top.
> * Add a `cancel` button. This button should redirect the user back to the accommodation they were editing.
> * Make the `Edit` button on `viewing single accommodation` a link and move it to the extreme right preferably in line with the breadcrumbs
> * Raise your coverage to 90+ and check out the other comments I have left on this pr

Noted. 
- Coverage raised. 
- Added Cancel button. 
- Moved the edit link to far-right. 
- Moved image edit to the bottom of the page
